src/kvstore/comm.h

```diff
@@ -53,13 +74,31 @@ class Comm {
    */
   virtual const NDArray& Reduce(
       int key, const std::vector<NDArray>& src, int priority) = 0;
+
   /**
    * \brief copy from src to dst[i] for every i
    */
   virtual void Broadcast(
       int key, const NDArray& src,
       const std::vector<NDArray*> dst, int priority) = 0;
 
+#if MXNET_USE_NCCL
+  // Aggregated reductions
+  virtual void Reduce(const std::vector<int> keys,
```

Comment:
If NCCL is going to do everything differently then it shouldn't inherit the comm interface. Do this in KVStoreNCCL directly

Reply:
Ok, will do.